src/consumers.ts

```diff
@@ -129,31 +150,46 @@ class SecurityEngine extends AnalysisConsumer implements DiagnosticProducer
 {
     constructor(public context: IDependency, config: any) {
         super(config);
-        this.binding = {path: ['result', 'recommendation', 'component-analyses', 'cve']};
+        this.binding = {path: ['component_analyses', 'vulnerability']};
         /* recommendation to use a different version */
-        this.changeToBinding = {path: ['result', 'recommendation', 'change_to']};
+        this.changeToBinding = {path: ['recommended_versions']};
+        /* snyk registration link */
+        this.registrationLinkBinding = {path: ['registration_link']};
+        /* Diagnostic message */
+        this.messageBinding = {path: ['message']};
+        /* Publicly known Security Vulnerability count */
+        this.vulnerabilityCountBinding = {path: ['known_security_vulnerability_count']};
+        /* Private Security Advisory count */
+        this.advisoryCountBinding = {path: ['security_advisory_count']};
     }
 
     produce(ctx: any): Diagnostic[] {
         if (this.item.length > 0) {
-            let cveList = [];
-            for (let cve of this.item) {
-                cveList.push(cve['id'])
+            /* DocumentUri as a URI object to pass URLs */
+            this.registrationLink as DocumentUri;
```

Comment:
Do you really need this?

Reply:
Not now, will need it when LSP is updated.
I will remove it as of now.

Reply:
@arajkumar PTAL : https://github.com/fabric8-analytics/fabric8-analytics-lsp-server/commit/0391ba4a038ba374107d78e1952d7d2b92367ab8